iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/unique_ptr.inl

```diff
@@ -25,148 +25,111 @@ namespace iox
 namespace cxx
 {
 template <typename T>
-unique_ptr<T>::unique_ptr(T* const ptr, const function<void(T*)>& deleter) noexcept
-    : m_ptr(ptr)
+inline unique_ptr<T>::unique_ptr(T* const object, const function<void(T*)>& deleter) noexcept
+    : m_ptr(object)
     , m_deleter(std::move(deleter))
 {
+    Ensures(object != nullptr);
```

Comment:
Why not use `not_null`?

Reply:
I think `not_null` is not the perfect type here. @elBoberido and me thought about a type which signals ownership like `Owns` or so but for now, keep it as minimal as possible and merge this thing.

Reply:
@elfenpiff Okidoki, no biggie. Can you elaborate your thoughts about `Owns` a bit more and/or provide and example?

Reply:
It originated in a comment in this PR: https://github.com/eclipse-iceoryx/iceoryx/pull/1644#discussion_r974315972

Reply:
@mossmaurice the core guidelines talk about and `owned<T*>` in cases one does not use a `unique_ptr`. It might be slightly better than `not_null`. It might be able to implement an `owned<T*>` and even a `optional<owned<T*>>` as efficient as a raw pointer but with the disadvantage that we must rely on the compiler/clang-tidy to detect use after move.